### PR TITLE
Adding an analyzer result object to API client.

### PR DESCRIPTION
**Pull request by kiddinn**
This PR adds a simple analyzer result object to the API client. When running an analyzer instead of returning back a JSON blob with the session ID an object is returned, which has functions to fetch the status of the analyzer being run, as well as getting the results back.

### CODE REVIEW

File: api_client/python/timesketch_api_client/analyzer.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+            return {}
+
+        data = response.json()
+
+        objects = data.get('objects')
+        if not objects:
+            return {}
+
+        for result in objects[0]:
+            result_id = result.get('analysissession_id', -1)
```
Comment (berggren) [inline api_client/python/timesketch_api_client/analyzer.py:50]: -1 ?

Comment (berggren) [reply api_client/python/timesketch_api_client/analyzer.py:50]: Why not None?

File: api_client/python/timesketch_api_client/error.py [hunk 2]
```diff
@@ -27,3 +27,11 @@ def error_message(response, message=None, error=RuntimeError):
         text = soup.p.string
     raise error('{0:s}, with error [{1:d}] {2!s} {3:s}'.format(
         message, response.status_code, response.reason, text))
+
+
+class Error(Exception):
```
Comment (berggren) [inline api_client/python/timesketch_api_client/error.py:32]: Pretty generic name. Maybe ApiClientError?

File: api_client/python/timesketch_api_client/analyzer.py (hunk 1 above)
Comment (kiddinn) [reply api_client/python/timesketch_api_client/analyzer.py:50]: No particular reason, other than to get an integer, might as well just have it default to none

File: api_client/python/timesketch_api_client/error.py (hunk 2 above)
Comment (kiddinn) [reply api_client/python/timesketch_api_client/error.py:32]: this is not an error to use... that iis this is following "proper way of doing things", see: https://docs.python.org/3/tutorial/errors.html#user-defined-exceptions

```
When creating a module that can raise several distinct errors, a common practice is to create a base class for exceptions defined by that module, and subclass that to create specific exception classes for different error conditions:
```

This is how we have done things in other projects, eg. plaso, etc

Comment (berggren) [reply api_client/python/timesketch_api_client/error.py:32]: okm

**Pull request merged**